### PR TITLE
HTS-548: Fixed bug for scenario when nino is not found so user isn't …

### PR DESCRIPTION
**Pull request by Jackie1988**
…eligible

### CODE REVIEW

File: app/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnector.scala [hunk 1]
```diff
@@ -45,16 +48,18 @@ class EligibilityCheckConnectorImpl @Inject() (http: WSHttp, metrics: Metrics, p
...
+    EitherT[Future, String, Option[EligibilityCheckResult]](
       {
         val timerContext = metrics.itmpEligibilityCheckTimer.time()
 
         http.get(url(nino), desHeaders)(hc.copy(authorization = None), ec)
           .map { response ⇒
             val time = timerContext.stop()
 
-            response.status match {
+            val r: Option[Either[String, EligibilityCheckResult]] = response.status match {
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnector.scala]: `r` isn't very descriptive - `result` maybe?

File: app/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnector.scala [hunk 2]
```diff
@@ -65,17 +70,22 @@ class EligibilityCheckConnectorImpl @Inject() (http: WSHttp, metrics: Metrics, p
                 }, _ ⇒
                   logger.info(s"Call to check eligibility successful, received 200 (OK) ${timeString(time)}", nino)
                 )
-                result
+                Some(result)
+
+              case Status.NOT_FOUND ⇒
+                pagerDutyAlerting.alert("Retrieved nino has not been found in DES, so user is not receiving Working Tax Credit")
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnector.scala]: do we really want a pager duty alert for this? It isn't an error per se is it?

File: test/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnectorSpec.scala [hunk 3]
```diff
@@ -55,7 +55,7 @@ class EligibilityCheckConnectorSpec extends TestSupport with GeneratorDrivenProp
     "return with the eligibility check result unchanged from ITMP" in {
       forAll { result: EligibilityCheckResult ⇒
         mockGet(connector.url(nino))(Some(HttpResponse(200, Some(Json.toJson(result))))) // scalastyle:ignore magic.number
-        Await.result(connector.isEligible(nino).value, 5.seconds) shouldBe Right(result)
+        Await.result(connector.isEligible(nino).value, 5.seconds) shouldBe Right(Some(result))
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnectorSpec.scala]: are there a unit test to test that a 404 maps onto `Right(None)`?

File: app/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnector.scala (hunk 2 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnector.scala]: No we don't, sorry that was an overlook

File: app/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnector.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnector.scala]: changed to res, as already have a val result in scope

File: test/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnectorSpec.scala (hunk 3 above)
Comment (AndyHWChung) [inline test/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnectorSpec.scala]: looks like you're still missing a unit test for the case when DES returns a 404 to this connector

Comment (Jackie1988) [reply test/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnectorSpec.scala]: Done

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**